executing/_position_node_finder.py

```diff
@@ -1,28 +1,30 @@
 import ast
 import dis
-from .executing import NotOneValueFound, only, function_node_types, assert_
+from enum import EnumMeta
+from types import CodeType, FrameType
+from typing import Any, Callable, Generator, Optional, Sequence, Tuple, Type, Union, cast
+from .executing import EnhancedAST, NotOneValueFound, Source, only, function_node_types, assert_
 from ._exceptions import KnownIssue, VerifierFailure
 
 from functools import lru_cache
 
 # the code in this module can use all python>=3.11 features
 
 
-def parents(node):
+def parents(node: EnhancedAST) -> Generator[EnhancedAST, None, None]:
```

Comment:
```suggestion
def parents(node: EnhancedAST) -> Iterator[EnhancedAST]:
```

Reply:
I've redone this one and all the other Generator usages as Iterator, because they were all the `[X, None, None]` case